### ISSUE TITLE
REFRESH MV panics with "frontier advancements after the `until` should be suppressed"

### ISSUE THREAD

**Issue by ggevay** [MaterializeInc/materialize#24481]
### What version of Materialize are you using?

278813c967bd9434af6c528ed4940a41093f2aac

### What is the issue?

We are hitting the `soft_panic` in the roundup operator of an MV that has two `REFRESH AT`s in the future. Repro [here](https://github.com/ggevay/materialize/tree/refresh-frontier-past-until-debugging). Run `materialized_views.slt`. The branch differs from main only in the minimization of the slt and some debug prints.

Weirdly, the error is not happening if I supply the same commands as the slt does, but through a psql shell.

**Comment by ggevay** [comment]
Turns out this is a new bug, because it's not happening at 815f7008524089e84b84ddb9c4d4e10ec7c58dc5. I'll do a git bisection now.

**Comment by ggevay** [comment]
It's 86b02f2cd81aa988eb71c28cd01832ec3c021521. In particular, this change
```
-            // TODO(txn-lazy): Get "lazy" flipped on before turning "lazy" on in prod.
-            persist_txn_tables_cli: Some(PersistTxnTablesImpl::Eager),
+            persist_txn_tables_cli: Some(PersistTxnTablesImpl::Lazy),
```
in the slt runner configuration. This also explains why it's happening only if I run these commands in an slt, and not through psql.

@danhhz

**Comment by ggevay** [comment]
Slack discussion: https://materializeinc.slack.com/archives/C01CFKM1QRF/p1705490015764119

**Comment by danhhz** [comment]
Assigning back to @ggevay for verification of the fix.

This only affects persist-txn=lazy, which is not on in prod, so pulling off the release blocker label.

**Comment by ggevay** [comment]
Umm, I'm still running into the same soft_panic. Sorry, I should have tested this before you merged your PR... @danhhz 

But anyhow, I'm probably going to just remove the `soft_panic` from the refresh code, so it's not a big problem. I thought through the refresh code again, and it really should be able to already handle a frontier advancement past the `until` just fine. Also, the [`until` discussion in the Timely channel](https://materializeinc.slack.com/archives/CMNT2QSBZ/p1705589936930539?thread_ts=1705509940.786839&cid=CMNT2QSBZ) has only 1 comment so far, and it is against including the suppressing of frontier advancements in the contract of `until`.

Btw. this is not a release blocker, because there is also another bug (#24469) that will make `REFRESH AT` not work in this release, and that bug is almost certainly not getting fixed today.

**Comment by danhhz** [comment]
Yeah, I wasn't 100% sure that was going to work, because of the 5, 7, 10 example I posted. As mentioned in slack, yeah, I think the assert is probably overly aggressive.

**Comment by ggevay** [comment]
Removed the soft_panic in https://github.com/MaterializeInc/materialize/pull/24582
